pkg/querier/querier.go

```diff
@@ -49,6 +50,7 @@ type Config struct {
 	Engine                        logql.EngineOpts `yaml:"engine,omitempty"`
 	MaxConcurrent                 int              `yaml:"max_concurrent"`
 	QueryStoreOnly                bool             `yaml:"query_store_only"`
+	QueryMemoryOnly               bool             `yaml:"query_memory_only"`
```

Comment:
I think calling it `query_ingester_only` would be clearer and matches what we are trying to do here.
```suggestion
	QueryIngesterOnly               bool             `yaml:"query_ingester_only"`
```